charts/flux2/templates/crd-controller-clusterrole.yaml

```diff
@@ -13,21 +13,31 @@ metadata:
     app.kubernetes.io/part-of: flux
     helm.sh/chart: {{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}
 rules:
+{{- if .Values.sourceController.create }}
```

Comment:
This is not Ok, as all other controllers need access to sources.

Reply:
My assumption was that the source controller will also be installed via the helm chart in that case and therefore this flag will be true. 

Do we also facilitate cases where a subset of the controllers is installed through the helm chart and others are installed differently? Cause if that is the case then this PR needs to be re-assessed ( from my side )